### DISCUSSION TITLE
stabilizer loop rate is off

### DISCUSSION THREAD

**Discussion by whY-note** [TinyMPC/TinyMPC#44] [Q&A] [answered]
When I run the tinympc code, it show that "STAB: WARNING: stabilizer loop rate is off ".
How to solve this problem?

**Comment by xkhainguyen** [comment, marked as answer]
Hi, thanks for reaching out!

We pushed the limits of the Crazyflie’s MCU, operating near the system’s boundary. As a result, it might disrupt the stabilizer task (which runs at the default 1000 Hz) when TinyMPC runs at higher rates or when other tasks, such as the state estimator or communication routines, are active.

In many cases, the Crazyflie still operates fine despite the warning. We often disable unnecessary tasks or routines in the firmware. In some experiments, we run TinyMPC at much lower rates within a cascaded control architecture. Feel free to ask more about the [TinyMPC-integrated firmware.](https://github.com/RoboticExplorationLab/tinympc-crazyflie-firmware/tree/main).
